src/preferences/colorpaletteeditormodel.h

```diff
@@ -46,3 +49,27 @@ class ColorPaletteEditorModel : public QStandardItemModel {
     bool m_bEmpty;
     bool m_bDirty;
 };
+
+class HotcueIndexListItem : public QStandardItem {
```

Comment:
The entire purpose of creating this class is to shift the "(de-)serialization" logic out of the model into the item so the item is responsible for translating between the String input and the data actually supposed to be represented as the string.